tests/test_label_categories.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+
+from datumaro import LabelCategories
+
+
+def test_add_category():
+    categories = LabelCategories()
+    index = categories.add("cat")
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].name == "cat"
+
+
+def test_add_category_with_parent():
+    categories = LabelCategories()
+    index = categories.add("cat", parent="animal")
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].name == "cat"
+    assert categories[0].parent == "animal"
+
+
+def test_add_category_with_attributes():
+    categories = LabelCategories()
+    attributes = {"color", "size"}
+    index = categories.add("cat", attributes=attributes)
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].attributes == attributes
+
+
+def test_add_duplicate_category():
+    categories = LabelCategories()
+    categories.add("cat")
+    with pytest.raises(KeyError):
+        categories.add("cat")
+
+
+def test_potential_collision():
+    """
+    Previously indices were computed as (parent or "") + name
+    """
```

Comment:
It's better to change the description to include information about the fixing PR and the problem, as this test checks for regressions in a specific use case.

Reply:
Included a link to the PR